<launch>
    <env name="GAZEBO_MODEL_PATH" value="$(find sm_fetch_two_table_whiskey_pour)/simulation/models/:$(find fetch_gazebo)/models:$(optenv GAZEBO_MODEL_PATH)" />
    <arg name="debug" default="false"/>
    <arg name="gui" default="true"/>
    <arg name="xterm_prefix" default="xterm -xrm 'XTerm*scrollBar:  true' -xrm 'xterm*rightScrollBar: true' -sl 10000 -hold -e "/>

    <env name="ROSCONSOLE_CONFIG_FILE" value="$(find sm_fetch_two_table_whiskey_pour)/config/rosconsole.config" />

    <!--  setup smacc planners configuration  -->
    <group ns="move_base">
        <!-- backward local planner -->
        <rosparam file="$(find sm_fetch_two_table_whiskey_pour)/config/move_base_client/backward_local_planner.yaml" command="load" />
        <!-- forward local planner -->
        <rosparam file="$(find sm_fetch_two_table_whiskey_pour)/config/move_base_client/forward_local_planner.yaml" command="load" />
    </group>

    <rosparam command="load" file="$(find sm_fetch_two_table_whiskey_pour)/config/sm_fetch_two_table_whiskey_pour_config.yaml" />
    <node pkg="sm_fetch_two_table_whiskey_pour" type="sm_fetch_two_table_whiskey_pour_node" name="sm_fetch_two_table_whiskey_pour_node" launch-prefix="$(arg xterm_prefix)" unless="$(arg debug)"/>

    <!-- Start Gazebo with a blank world -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="debug" value="$(arg debug)" />
        <arg name="gui" value="$(arg gui)" />
        <arg name="paused" value="false"/>
        <arg name="use_sim_time" value="true"/>
        <arg name="headless" value="false"/>
        <arg name="world_name" value="$(find sm_fetch_two_table_whiskey_pour)/simulation/worlds/test_zone_pick_place.sdf"/>
    </include>

    <include file="$(find fetch_moveit_config)/launch/move_group.launch">
        <arg name="pipeline" value="ompl"/>
    </include>

    <node name="$(anon rviz)" pkg="rviz" type="rviz" respawn="false" args="-d $(find sm_fetch_two_table_whiskey_pour)/config/rviz.rviz" output="screen">
        <rosparam command="load" file="$(find sm_fetch_two_table_whiskey_pour)/config/manipulation/kinematics.yaml"/>
    </node>

    <!--················ NAV ·······················-->
    <include file="$(find sm_fetch_two_table_whiskey_pour)/launch/include/sm_fetch_two_table_whiskey_pour_nav.launch"/>

    <!-- Oh, you wanted a robot? -->
    <include file="$(find sm_fetch_two_table_whiskey_pour)/launch/include/fetch.launch.xml" />

</launch>
